feat: add support for GitLab Generic Packages

neomantra opened the pull request:
GitLab's Generic Package Repository allows uploading files
to GitLab, as an alternative to attachments.

Added the flag `use_package_registry` to the `gitlab_urls` config.
When false, the default, it will behave as before.
When true, it will publish the file to the Generic Package Registry.
Either way, the URL is bound to the release the same

Updated `go-gitlab` to version 0.52.2 which has the new
`GenericPackages.PublishPackageFile` function.

----
Fixes #2006.  

I tried to run local tests on it and it was not working (failures with `internal/pipe` and weird assembly output);  however, I think that is unrelated to this PR as I've tested this many times before without issue.  My apologies if this is not the case and the CI fails.

caarlos0 commented on www/docs/scm/gitlab.md:
````diff
@@ -37,6 +40,14 @@ If none are set, they default to GitLab's public URLs.
+
+GitLab introduced the [Generic Package Registry](https://docs.gitlab.com/ee/user/packages/package_registry/index.html) in Gitlab 13.5.  Normally, `goreleaser` uploads release files as "attachments", which may have [administrative limits](https://docs.gitlab.com/ee/user/admin_area/settings/account_and_limit_settings.html).  Notably, hosted gitlab.com instances have a 10MB attachment limit which ca ...
+
+```
````
```suggestion
```yaml
# .goreleaser.yml
```

caarlos0 commented on www/docs/scm/gitlab.md:
```diff
@@ -37,6 +40,14 @@ If none are set, they default to GitLab's public URLs.
 
+## Generic Package Registry
+
+GitLab introduced the [Generic Package Registry](https://docs.gitlab.com/ee/user/packages/package_registry/index.html) in Gitlab 13.5.  Normally, `goreleaser` uploads release files as "attachments", which may have [administrative limits](https://docs.gitlab.com/ee/user/admin_area/settings/account_and_limit_settings.html).  Notably, hosted gitlab.com instances have a 10MB attachment limit which ca ...
```
```suggestion
GitLab introduced the [Generic Package Registry](https://docs.gitlab.com/ee/user/packages/package_registry/index.html) in Gitlab 13.5.
Normally, `goreleaser` uploads release files as "attachments", which may have [administrative limits](https://docs.gitlab.com/ee/user/admin_area/settings/account_and_limit_settings.html).  Notably, hosted gitlab.com instances have a 10MB attachment limit which cannot be changed.
Uploading to the Generic Package Registry does not have this restriction.  To use it instead, set `use_package_registry` to `true`.
```

neomantra replied:
Sure, changing this now -- did you make (and want)  the `not`  in line 47 the color red somehow or is that just how GitHub is showing it?

neomantra replied:
OK pushed that (without anything with `not` coloring).

caarlos0 reviewed (commented):
looking good overall, added just a couple of comments

neomantra commented:
I have rebased this PR to main and squashed the doc PR commit.   
There are less file changes now (e.g. `go.mod`) because `go-gitlab` was upgraded to `0.52.2` in  2a340e9125f417495eac191af03c880e61004344.

caarlos0 commented:
LGTM! Thanks for the hard work on this!
